bridgetown-core/lib/bridgetown-core/front_matter/loaders.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Bridgetown
+  module FrontMatter
+    module Loaders
+      autoload :Base, "bridgetown-core/front_matter/loaders/base"
+      autoload :Ruby, "bridgetown-core/front_matter/loaders/ruby"
+      autoload :YAML, "bridgetown-core/front_matter/loaders/yaml"
+
+      Result = Struct.new(:content, :front_matter, :line_count, keyword_init: true)
+
+      # Constructs a list of possible loaders for a {Model::RepoOrigin} or {Layout}
+      #
+      # @param origin_or_layout [Bridgetown::Model::RepoOrigin, Bridgetown::Layout]
+      # @return [Array<Loaders::Base>]
+      def self.for(origin_or_layout)
+        registry.map { |loader_class| loader_class.new(origin_or_layout) }
+      end
+
+      # Determines whether a given file has front matter
+      #
+      # @param path [Pathname, String] the path to the file
+      # @return [Boolean] true if the file has front matter, false otherwise
+      def self.front_matter?(file)
+        registry.any? { |loader_class| loader_class.header?(file) }
+      end
+
+      # Registers a new type of front matter loader
+      #
+      # @param loader_class [Loader::Base]
+      # @return [void]
+      def self.register(loader_class)
+        registry.push(loader_class)
+      end
+
+      private_class_method def self.registry
+        @registry ||= []
+      end
+
+      register YAML
+      register Ruby
```

Comment:
**comment:** This is at MVP stage and it's what I'm least excited about. If you notice, I register YAML before Ruby. This is because it needs to be higher precedence to maintain the original business logic of the `read_from_matter` method.

I didn't want to overengineer here, but I think having both (1) symbol-based registration and (2) a priority level would be useful here. That way people can override by type and give higher precedence to a loader if needed.

Thoughts?